src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/DeliverymenRequestResolver.java

```diff
@@ -203,4 +201,64 @@ public MutableComponent getRequesterDisplayName(
     {
         return Component.translatable(TranslationConstants.COM_MINECOLONIES_COREMOD_JOB_DELIVERYMAN);
     }
+
+    /**
+     * Try to rebalance the queues of current dman requests linked to this warehouse
+     */
+    public void rebalanceCurrentRequests(@NotNull final IRequestManager manager)
+    {
+        // ideally we'd do a little work each tick to spread out the load, but sadly the colony tick only happens every ~25s
+        final Set<IRequest<?>> alreadyMoved = new HashSet<>();
+        final List<ICitizenData> workers = getResolveAbleDeliverymen(manager);
+        for (final ICitizenData worker : workers)
+        {
+            final JobDeliveryman workerJob = (JobDeliveryman) worker.getJob();
+            final List<IToken<?>> taskQueue = workerJob.getTaskQueue();     // assumed immutable copy, not live list
+            final boolean isWorking = worker.isWorking() && !worker.isPaused();
+
+            // don't attempt to reassign the current task unless the worker isn't actually working
+            for (int requestIndex = isWorking ? 1 : 0; requestIndex < taskQueue.size(); ++requestIndex)
```

Comment:
Seems complicated, why not just if notWorking, return?

Reply:
I think rebalancing 1 task every 25seconds is also okay, tasks usually need around that to be done anyway

Reply:
> Seems complicated, why not just if notWorking, return?

Because we **do** want to go through all the requests of the non-working dmen.  We just skip the first request of the working one because they're probably out doing it right that moment and we don't want to rugpull them.

Reply:
> I think rebalancing 1 task every 25seconds is also okay, tasks usually need around that to be done anyway

I was more sort of talking about how it seemed a little odd that the whole `colonyStateMachine` thing appears to be set up to do each task on a different tick to spread out the processing, but then `worldTickSlow` calls about fifteen different things all on the same tick...

Reply:
its because inherently worldTickSlow only had light updates... and the biggest point is to spread all colonies from doing it on the same tick... Also do you see how using ... gives a neat impression of passive agressiveness?... I'd suggest we do not use them...